pyDmdReader/_dmd_reader.py

```diff
@@ -203,10 +205,13 @@ def read_array(
         for ch_name in used_channels:
             ch_config = self.__channels[ch_name]
 
-            if start_time == 0.0 and end_time is None:
-                timestamps, data = self.__get_all_sweeps(ch_config, max_samples)
+            if start == 0 and end is None:
+                timestamps, data = self.__get_all_sweeps(ch_config)
             else:
-                timestamps, data = self.__get_ranged_sweeps(ch_config, start_time, end_time, max_samples)
+                if by_index:
```

Comment:
We probably should make sure all channels have the same sample rate, otherwise we mix indices from different timestamps. I can take a look at that when adding unit tests later on.